### DISCUSSION TITLE
AutoDisposeStateNotifierProvider issue.

### DISCUSSION THREAD

**Discussion by Patrick386** [rrousselGit/riverpod#721] [Q&A]
AutoDisposeStateNotifierProvider issue.
State values are constantly initialized.
You can tell it's initialized when you get data from Firebase and store it in a state and other widgets look at the state value.
It behaves differently than before migration.

**Comment by rrousselGit** [comment]
I do not understand the problem.

Could you share some code to reproduce the issue?

**Comment by Patrick386** [comment]
riverpod 0.14.0+3
- It works fine.

```
final pageNotifierProvider =
StateNotifierProvider.autoDispose<PageNotifier, PageState>( (AutoDisposeProviderReference ref) {

     return PageNotifier(ref.read);
   });
```

New Migration code.
- For some reason the state keeps being initialized.

```
final  AutoDisposeStateNotifierProvider<PageNotifier, PageState> pageNotifierProvider =
   StateNotifierProvider.autoDispose <PageNotifier, PageState>(
       ( AutoDisposeStateNotifierProviderRef ref) {

 return PageNotifier(ref.read);
});
```

My code is complex.
After more testing, I will post the reproducible sample code again.

**Comment by aqwert** [comment]
From what I understand, when you have an `autoDispose` provider and you call `.read` on it what will happen is the provider will run but because there are no listeners it will dispose straight away. 

I discovered this also and now where ever I have an `autoDispose` provider I make sure there is a `watch` on it at the right place to keep it alive for the duration of use.
